app/src/main/java/com/example/habitsmasher/EmailValidator.java

```diff
@@ -0,0 +1,44 @@
+package com.example.habitsmasher;
+
+import android.util.Patterns;
+import android.widget.EditText;
+
+import androidx.core.util.PatternsCompat;
+
+
+/**
+ * This class validates whether a email is of the correct format.
+ * This class is based of the UserValidator class.
+ */
+public class EmailValidator {
```

Comment:
It's good that we created a new validator to check the emails, only thing we need to do now is actually use this new validator in the `UserValidator` class! 

Let's replace and update all the email validation logic in that class to use this new validator class instead.

It's good that we have UI test coverage for this workflow, but we are also missing **unit tests** for this specific validator... Let's create some unit tests to test the `isEmailValid` method:
* valid email, return true
* invalid email, return false
* empty email, return false

- [ ] update email validation in `UserValidator` to use `EmailValidator`
- [ ] create new unit tests for `EmailValidator`
- [ ] make sure all old/new unit tests pass after this change